### PR TITLE
Signup 0.1

### PR DESCRIPTION
**Pull request by ellio**
Added:
- Basic signup functionality
- Requires name, email, password, and username.  Confirmations of obvious fields
- Password must be 6 characters or more, contain one lower-case, one upper-case, and one non-letter character
- Username must be 4 characters or more
- Email must pass standards
- All fields checked once left, messages shown with ajax if they are invalid (or if a username/email is already in use)
- If submitted, fields are again checked and errors will be shown in red

Major missing:
- YOU HAVE TO BE SIGNED IN TO REGISTER.  This is an obvious flaw :p.  If you have any idea what I need to do to make this page available, let me know.
- Email confirmation

### CODE REVIEW

File: app/controllers/Signup.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+@With(Secure.class)
+public class Signup extends OBController {
+	
+	public static void signup() {
+		render();
+	}
+	
+	public static void signup_user(String firstName, String lastName, String username, String email, String email2, String password, String password2) {
+		validation.email(email).message("Please use a valid email address");
```
Comment (amshali) [inline app/controllers/Signup.java:17]: I think you can use java annotations instead. But this is good too. Try that later.

File: app/controllers/Signup.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		validation.isTrue(lastName.trim().length() >= 1).message("Please enter a last name");
+		username = username.trim();
+		validation.isTrue(username.trim().length() >= 4).message("Username must be at least 4 characters");
+		validation.isTrue(password.length() >= 6).message("Password must be at least 6 characters");
+		validation.match(password, ".*[A-Z]+.*").message("Password must contain an uppercase letter");
+		validation.match(password, ".*[a-z]+.*").message("Password must contain a lowercase letter");
+		validation.match(password, "(.*\\W+.*)|(.*[0-9]+.*)").message("Password must contain a non-letter character");
+		validation.equals(password, password2).message("Passwords do match");
+		if (validation.hasErrors()) {
+			String errors ="";
```
Comment (amshali) [inline app/controllers/Signup.java:31]: There might be a better way for doing this. Checkout the play tutorial.

File: app/views/Signup/signup.html [hunk 3]
```diff
@@ -0,0 +1,166 @@
+<!DOCTYPE html >
```
Comment (amshali) [inline app/views/Signup/signup.html:1]: There must be a link to this page from somewhere.

File: app/controllers/Signup.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
+package controllers;
+
+import java.util.*;
+import models.User;
+import play.data.validation.Error;
+import controllers.Secure;
+import play.mvc.With;
+
+@With(Secure.class)
+public class Signup extends OBController {
```
Comment (jtessler) [inline app/controllers/Signup.java:10]: I think the problem is the `@With(Secure.class)`. You also extend from `OBController`, which also has the security annotation. I think if you remove the annotation here and extend from `Controller` instead, it'll work.

**Pull request merged**